tests/repoproviders_test.py

```diff
@@ -0,0 +1,58 @@
+from unittest import TestCase
+
+import pytest
+
+from binderhub.repoproviders import GitHubRepoProvider
+
+
+# General string processing
+@pytest.mark.parametrize(
+    'raw_text, suffix, clean_text', [
+        ("foo.git", ".git", "foo"),
+        ("foo.bar", ".git", "foo.bar"),
+        ("foo.bar", ".bar", "foo")
+    ]
+)
+def test_string_strip(raw_text, suffix, clean_text):
+    assert GitHubRepoProvider.strip_suffix(raw_text, suffix) == clean_text
+
+
+# user/repo/reference
+@pytest.mark.parametrize(
+    'spec, raw_user, raw_repo, raw_ref', [
+        ("user/repo/master", "user", "repo", "master"),
+        ("user/repo/hotfix/squash-bug", "user", "repo", "hotfix/squash-bug"),
+        ("user/repo/feature/save_world", "user", "repo", "feature/save_world")
+    ]
+)
+def test_spec_processing(spec, raw_user, raw_repo, raw_ref):
+    user, repo, unresolved_ref = GitHubRepoProvider.process_spec(spec)
+    assert raw_user == user
+    assert raw_repo == repo
+    assert raw_ref == unresolved_ref
+
+
+class TestSpecErrorHandling(TestCase):
```

Comment:
What does using `TestCase` do for us? Asking because I've never had the need (or known) to use one before with pytest.

Reply:
I use it mainly because I like `assertRaisesRegexp`, but it could go away if that gets replaced with pytest.raises.

In more complex situations, it also gives you a convenient place to define `setup` and `teardown` methods which will automatically run before and after a group of tests.